### ISSUE TITLE
Thinstation printers

### ISSUE THREAD

**Issue by RichCoop** [Thinstation/thinstation#699]
Hello, i try to make the terminal server grab the lan printer, but it doesn't work.
I connect to to it through freerdp.

I write this in the thinstation.conf.buildtime:

CUPS_PRINTER_0_NAME="Kyocera TASKalfa 3500i"
CUPS_PRINTER_0_DEVICEURI="socket://192.168.0.3"
CUPS_PRINTER_0_DRIVER="drv:///sample.drv/generpcl.ppd"
and 
SESSION_0_FREERDP_OPTIONS="/f /v /d /cert-ignore /g /printer"

What do i do wrong?
P.S. i don't specify the /printer parameter as later i will add more printers and i need them all to be available. According to the xfreerdp keys, if i write just '/printer' it should take all the printers.

**Comment by RichCoop** [comment]
I tried to specify the printer and still nothing.

**Comment by RichCoop** [comment]
Maybe something else is required? Or i write these lines wrong?

**Comment by Thinstation** [comment]
Does the printer get mapped in CUPS with those lines? Check at http://localhost:631

**Comment by RichCoop** [comment]
I should follow localhost from the inside of thinstaion? Sorry for stupid questions, I'm just kinda new to this.

**Comment by Thinstation** [comment]
Yes, you need to open a web browser and go to http://localhost:631 on a thinstation.

**Comment by RichCoop** [comment]
So i need to add a package with it to the build.conf? Or you meant open this from the terminal session?

**Comment by Thinstation** [comment]
Do a build with chrome, and then open it on the thinstation and confirm that your printer was mapped.

**Comment by RichCoop** [comment]
I added chrome package into the build.conf.
When i try to run it, thinstation says:
[0211/123110.157789:ERROR:file_writer.cc(98)] writev: No space left on device (28)
[0211/123110.157789:ERROR:crash_report_exception_handler.cc(226)] WriteEverything failed
Trace/breakpoint trap

rootfs is 100% used, i believe this is the problem, how can i add more space to it? The disk itself thinstation runs on is 250g.

Also I didn't include 'package lprng' in the build.conf because if i do that i receive Kernel panic error during the thinstation launch.

**Comment by Thinstation** [comment]
More RAM. Try developing a virtual machine first.

**Comment by RichCoop** [comment]
More ram? Isn't 4 gb enough for this?

**Comment by Thinstation** [comment]
If this happens when building, you need more disk space. If this happens
when booting, you need more ram.

**Comment by RichCoop** [comment]
Hmmm during the build stage all seems to be fine, got to check the process logs more carefully though. However it would be strange that 250gb of the disk space isn't enough for it.
As for the ram, i thought 4 gb is enough too, the problem is, it is a physical machine and 4gb is the max, we don't have any more ddr2. Can it be that with the chrome package some other packages are necessary? Also i checked df -h, here is what it shows:
Filesystem, size, used,available,use%,mounted on
rootfs 1.4G 1.4g 0 100% /
devtmpfs 1.4G 0 1.4G 0% /dev
tmpfs 1.7G 0 1.7G 0% /dev/shm
tmpfs 1.7G 932.0K 1.7G 0% /run
tmpfs 1.7G 0 1.7G 0% /sys/fs/cgroup
tmpfs 1.7G 16.0K 1.7G 0% /tmp 
/dev/sdb1 238.8G 568.1M 232.3G 0% /mnt/disc/sdb/part6
/dev/sda1 27.3G 4.6G 21.3G 0% /mnt/disc/sda/part1
/dev/sda6 199.7G 17.3G 172.2G 9% /mnt/disc/sda/part6
tmpfs 349.6M 4.0K 349.6M 0% /run/user/1000

**Comment by Thinstation** [comment]
rootfs 1.4G 1.4g 0 100% /
is full. free up some space on /

**Comment by RichCoop** [comment]
How should i do that?

**Comment by Thinstation** [comment]
You need more RAM. Develop on a VM. Once you have everything working, you can remove the things you don't need anymore and make your image small for real hardware.

**Comment by RichCoop** [comment]
I ve tired it on another physical machine, this time it has 8gb RAM, no error above, except it doesn't work either.
This time it only says 'Trace/breakpoint trap'

**Comment by RichCoop** [comment]
Perhaps i need to include more packages?

**Comment by RichCoop** [comment]
Other browsers don't work either.

**Comment by Thinstation** [comment]
The browser is the most tested application for ThinStation that there is. Maype Putin did something to your internet?

**Comment by RichCoop** [comment]
Unlikely, as we use proxy in Germany. I receive this error all the time:

'Trace/breakpoint trap'

I'm pretty sure i need some more packages to include, which ones should be 100% included, maybe i miss some?

**Comment by Thinstation** [comment]
You can use the /build/conf/basic/* files. Just uncomment chrome and build.

**Comment by RichCoop** [comment]
So, you suggest to replace my current ones with these? OK, i will try it.

**Comment by RichCoop** [comment]
Still the same, i think the reason is that these browsers require some graphics.
 I tried openbox, but it fails to run, it says:
 
 "Failed to open the display from the DISPLAY environment variable"

Same thing for xfmw4.

**Comment by Thinstation** [comment]
Are you doing an allmodules/allfirmware build with all the xorg7- (except nvidia)?

**Comment by RichCoop** [comment]
I picked all the xorg7 including nvidia and i build with --allmodules and --autodl.
The error is the same.

**Comment by Thinstation** [comment]
The nvidia xorg7 breaks everything, including the chroot. You have to nuke
and pave, and don’t ever pick the xorg7_nvida package.

**Comment by RichCoop** [comment]
I cleaned my chroot and did everything again, this time without the nvidia, no changes.

**Comment by Thinstation** [comment]
Idk. VM's with browsers are the most tested build ever. I made a whole video running through every step.

**Comment by RichCoop** [comment]
[build.conf.txt](https://github.com/Thinstation/thinstation/files/8130173/build.conf.txt)
[thinstation.conf.buildtime.txt](https://github.com/Thinstation/thinstation/files/8130174/thinstation.conf.buildtime.txt)

I tried on several physical machines, same thing. Maybe i miss something in my packages?

**Comment by Thinstation** [comment]
And when you build in a VM?

**Comment by RichCoop** [comment]
The vm doesn't launch i receive the error: "Kernel panic - not syncing: VFS: Unable to mount root fs on uknown-block (0,0)

After some adjustments it gives me this error instead: "/init: error while loading shared libraries: librt.so.1: cannot open shared object file: No such file or directory."

**Comment by RichCoop** [comment]
Dunno what to do now :/

**Comment by Thinstation** [comment]
Add more RAM

**Comment by RichCoop** [comment]
That was the soltution, but all the other problems above persist in the virutal machine aswell.

**Comment by Doncuppjr** [comment]
Bummer


Sent from Yahoo Mail for iPhone


On Monday, February 28, 2022, 6:56 AM, RichCoop ***@***.***> wrote:




That was the soltution, but all the other problems above persist in the virutal machine aswell.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by RichCoop** [comment]
With ram that was my fault yes, however i was pretty sure that 4 gb would be enough as it turned out i needed 6.
But still can't open chrome on the VM too, aswell as i can't setup printers.

**Comment by Thinstation** [comment]
If you still can’t open chrome with 6gb in a VM, you have downloaded hacked
binaries or have a bad internet connection.

**Comment by RichCoop** [comment]
It is highly unlikely as i downloaded and configured things directly from the links here. And i tried this on the physical machines and the virtual ones. I even tried a machine with 8 gb ram and still the same thing. Tried it in different networks and still the same thing. I believe i miss something in the configs, but i don't know what.

**Comment by RichCoop** [comment]
I ve tried this again, it looks like openbox works in the session where connection window pops up. But i can't open anything else there, if i open another terminal session, openbox doesn't work aswell as any browsers.
